ArtOfIllusion/src/artofillusion/SceneViewer.java

```diff
@@ -107,26 +106,46 @@ public void setOrientation(int which)
     super.setOrientation(which);
     if (which > 5 && which < 6+cameras.size())
     {
-		ObjectInfo nextCamera = cameras.elementAt(which-6);
-		CoordinateSystem coords = nextCamera.coords.duplicate();
-
-		if (nextCamera.getObject() instanceof SceneCamera){
-			nextCenter = new Vec3(coords.getOrigin().plus(coords.getZDirection().times(((SceneCamera)nextCamera.getObject()).getDistToPlane())));
-		}
-		else if (nextCamera.getObject() instanceof SpotLight)
-			nextCenter = new Vec3(coords.getOrigin().plus(coords.getZDirection().times(((SpotLight)nextCamera.getObject()).getDistToPlane())));
-		else if (nextCamera.getObject() instanceof DirectionalLight)
-			nextCenter = new Vec3(coords.getOrigin().plus(coords.getZDirection().times(((DirectionalLight)nextCamera.getObject()).getDistToPlane())));
-		else
-			return;
-
-		animation.start(coords, rotationCenter, 100.0, which, navigation);
+        ObjectInfo nextCamera = cameras.elementAt(which-6);
+        CoordinateSystem nextCoords = nextCamera.coords.duplicate();
+        double dtp, nextScale;
```

Comment:
What does dtp mean here? Please expand the abbreviations throughout this section, for clarity...

Reply:
OK. Did just this bit. 

There are a couple of almost similar ones elsewhere, but I did not feel like rocking the boat... The equations are the same but the content used varies just so slightly, that getting it right takes focus, that I may not have at this hour...

Reply:
Situations like this are, why I think, all camera-capable objects should be put under a common interface. Could just ask the bound object the parameter, that is needed without having to check several object types and in some cases calculate the needed number from other information. Most of this code would be compressed to a couple of lines.